resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -294,10 +295,15 @@ private[spark] class KubernetesClusterSchedulerBackend(
   }
 
   private class KubernetesDriverEndpoint extends DriverEndpoint {
+
+    protected val execIDRequester = new HashMap[RpcAddress, String]
```

Comment:
This looks like monotonically increasing and will cause OOM in JVM in case of dynamic allocation or executor rolling case. What do you think about that, @holdenk ?